packages/material-ui/src/Breadcrumbs/Breadcrumbs.js

```diff
@@ -26,99 +26,100 @@ const styles = {
+  const {
+    children,
+    classes,
+    className,
+    component: Component,
+    itemsAfterCollapse,
+    itemsBeforeCollapse,
+    maxItems,
+    separator,
+    ...other
+  } = props;
 
-  renderItemsBeforeAndAfter(allItems) {
-    const { maxItems, itemsBeforeCollapse, itemsAfterCollapse } = this.props;
+  const [expanded, setExpanded] = React.useState(false);
 
-    // This defends against someone passing weird input, to ensure that if all
-    // items would be shown anyway, we just show all items without the EllipsisItem
-    if (itemsBeforeCollapse + itemsAfterCollapse >= allItems.length) {
-      warning(
-        false,
-        [
-          'Material-UI: you have provided an invalid combination of properties to the Breadcrumbs.',
-          // eslint-disable-next-line max-len
-          `itemsAfterCollapse={${itemsAfterCollapse}} + itemsBeforeCollapse={${itemsBeforeCollapse}} >= maxItems={${maxItems}}`,
-        ].join('\n'),
-      );
-      return allItems;
-    }
+  const handleClickExpand = () => {
+    setExpanded(true);
+  };
 
-    return [
-      ...allItems.slice(0, itemsBeforeCollapse),
-      <BreadcrumbCollapsed key="ellipsis" onClick={this.handleClickExpand} />,
-      ...allItems.slice(allItems.length - itemsAfterCollapse, allItems.length),
-    ];
-  }
+  const renderItemsBeforeAndAfter = React.useMemo(
+    () => allItems => {
+      // This defends against someone passing weird input, to ensure that if all
+      // items would be shown anyway, we just show all items without the EllipsisItem
+      if (itemsBeforeCollapse + itemsAfterCollapse >= allItems.length) {
+        warning(
+          false,
+          [
+            // eslint-disable-next-line max-len
+            'Material-UI: you have provided an invalid combination of properties to the Breadcrumbs.',
+            // eslint-disable-next-line max-len
+            `itemsAfterCollapse={${itemsAfterCollapse}} +itemsBeforeCollapse={${itemsBeforeCollapse}} >= maxItems={${maxItems}}`,
+          ].join('\n'),
+        );
+        return allItems;
+      }
 
-  render() {
-    const {
-      children,
-      classes,
-      className: classNameProp,
-      component: Component,
-      itemsAfterCollapse,
-      itemsBeforeCollapse,
-      maxItems,
-      separator,
-      ...other
-    } = this.props;
+      return [
+        ...allItems.slice(0, itemsBeforeCollapse),
+        <BreadcrumbCollapsed key="ellipsis" onClick={handleClickExpand} />,
+        ...allItems.slice(allItems.length - itemsAfterCollapse, allItems.length),
+      ];
+    },
+    [handleClickExpand, itemsAfterCollapse, itemsBeforeCollapse, maxItems],
+  );
 
-    const allItems = React.Children.toArray(children)
-      .filter(child => React.isValidElement(child))
-      .map((child, index) => (
-        <li className={classes.li} key={String(index)}>
-          {child}
-        </li>
-      ));
+  const allItems = React.Children.toArray(children)
```

Comment:
Simpler:
```suggestion
  const allItems = React.Children.toArray(children)
    .filter(child => React.isValidElement(child))
    .map(child => <li className={classes.li}>{child}</li>);
```

`key={String(index)}` is a crazy hack that eslint isn't smart enough to handle. It's something I have used at multiple ocassions in the past instead of disabling the failing eslint rule. It doesn't change the code behavior. We can remove it.